runtime/util/hshelp.c

```diff
@@ -898,9 +898,18 @@ areMethodsEquivalentSub(J9ROMMethod * method1, J9ROMClass * romClass1, J9Class *
 					/* if RAM classes are not provided do not attempt to propagate resolved callsite */
 					if ((NULL != ramClass1) && (NULL != ramClass2)) {
 						/* propagate call site data */
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+						J9InvokeCacheEntry * entry1 = (J9InvokeCacheEntry *)ramClass1->callSites + callSiteIndex1;
+						if (NULL != entry1->target) {
```

Comment:
Is it valid to have a non-null `target` and a null `appendix`?  If not, I'd prefer to check for both here before propagating state.